src/Host.cpp

```diff
@@ -495,6 +496,8 @@ Host::Host(int port, const QString& hostname, const QString& login, const QStrin
         auto entry = i.next();
         profileShortcuts.insert(entry, new QKeySequence(*mudlet::self()->mShortcutsManager->getSequence(entry)));
     }
+
+    startMapAutosave();
```

Comment:
Is this the most appropriate point to initiate autosaving the map?

For a **new** profile (not a copy of an existing one) there won't necessarily be a map yet and for an existing one IIRC a map may not get loaded until the "Map" toolbar button is clicked for the first time if a mapper is not instantiated by a script/package...

_Looking at the calls to `new dlgMapper(...)` they happen in:_
* _`TMainConsole::createMapper(...)`: 2 alternatives one for the Main console and one for when it is in a sub-console)_
* _`Host::createMapper(...)`_

*Oh! That is messy/confusing - there are MORE calls to `TMap::restore(...)` than that - ah, that might explain why the map is sometimes loaded multiple times on start-up - something to investigate and fix I suppose.*


Also, in some extreme, pathological, cases it can take a long time to load and audit a map on a slow machine with a big map (I was testing something on a Windows machine with a debug build {so automatically slower} with that humongous >2M room map recently and that took nearly an hour to finish the loading!) Might it not be better to make `startMapAutosave()` **idempotent** and to call it after each call to TMap::audit()` - so it only starts once a map has been **successfully** loaded and audited - ah, that won't include the case where the user is starting from scratch.

Okay, how about changing the lines that do `mpMap->mUnsavedMap = true` to call a method that both sets the flag AND starts the autosave timer if it hasn't been started before?

What about the cases after `TMap::mapClear()` gets called - should that **stop** the timer (and also run the timer code) so that the (to be/just) deleted map is saved in the automatic backup until a new map overwrites it?

Reply:
@SlySven it's important to focus on keeping things as simple as possible instead of complicated! If you find any practical issues with autosaves, I'm happy to fix them, we can't afford to spend time and energy thinking of theoretical problems and then solving those theoretical problems.